src/mqtt.js

```diff
@@ -608,6 +608,28 @@ function convertHexIdToNumericId(hexId) {
     return BigInt('0x' + hexId.replaceAll("!", ""));
 }
 
+/**
+ * Obfuscates overly precise location data
+ * @param coordinate the original coordinate (lat or lng)
+ * @returns {coordinate} the obfuscated coordinate
+ */
+function obfuscateCoordinate(coordinate) {
+    // Leave this as a string rather than try to unnecessarily parseFloat
+    if (coordinate.includes(".")) {
+        const parts = coordinate.split(".");
+        let precisionPart = parts[1];
+        // This will catch anything 1.1km or more precise, or two decimal places
+        if (parts[1].length > 1) {
+            // Take only the first digit
+            precisionPart = precisionPart.substring(0, 1);
+            // Add a random digit for the 1.1km precision point
+            precisionPart += Math.floor(Math.random() * 10);
+        }
+        return parts[0] + "." + precisionPart;
+    }
+    return coordinate;
+}
```

Comment:
Why is this logic not behind a flag/option? 

Many folks (myself included) run local instances of this SW, and making this change unilaterally without it being opt-in by the operator (or exposing a way to disable the behavior) is a regression in functionality. I don't really like this change in principle, but if you're going to add it, put it behind a flag (preferably off by default).

Reply:
Done.

Reply:
Also added configurable precision.

Reply:
Of course! Your feedback made it better :-)